### PR TITLE
kubekins/krte: Add `main` variant

### PR DESCRIPTION
**Pull request by justaugustus**
- kubekins/krte: Add `main` variant (part of https://github.com/kubernetes/enhancements/issues/2853, cc: @cpanato)

Signed-off-by: Stephen Augustus <foo@auggie.dev>

/hold for https://github.com/kubernetes/kubernetes/pull/107105
/assign @cpanato @puerco 
cc: @kubernetes/release-engineering

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by BenTheElder** [comment]
I missed, have we tested on -experimental at all yet? 1.18 looks like it has some exciting / major changes :D

**Comment by justaugustus** [comment]
https://github.com/kubernetes/test-infra/pull/25723 bumped the golang version to go1.18, so I've updated this to only include the new `main` variant.
@kubernetes/release-engineering -- Needs `/lgtm`!

> I missed, have we tested on -experimental at all yet? 1.18 looks like it has some exciting / major changes :D

@BenTheElder -- We created the `go-canary` variant in https://github.com/kubernetes/test-infra/pull/18328 to prevent disruption for other experimental jobs when changing Golang major versions.

The `experimental` variant is used in enough non-SIG Release/Testing, that I've never felt comfortable with subjecting that many jobs to the change.

So right now, if someone is feeling adventurous: use `go-canary`.
If not, they can use the `master` variant which doesn't get moved to "next" Golang until it lands in k/k.

That said, we don't really have a policy re: variants.
@kubernetes/sig-testing-leads -- WDYT?

**Comment by BenTheElder** [comment]
I'm not sure what happened where this go round, but the go1.18 update was extremely not smooth, it doesn't feel like it got soaked in CI ahead of updating.

**Comment by liggitt** [comment]
if the intent of this is just to enable master/main kubekins tags to be identical, this seems harmless.

I agree we should stop bumping master (and now main) images to a new version of go before demonstrating green runs on the go-canary / experimental image, but that seems outside the scope of this PR.

@justaugustus, is there an issue tracking the fallout of the last CI go bump and how we'll change the rollout for go 1.19?

**Comment by justaugustus** [comment]
> I'm not sure what happened where this go round, but the go1.18 update was extremely not smooth, it doesn't feel like it got soaked in CI ahead of updating.

> I agree we should stop bumping master (and now main) images to a new version of go before demonstrating green runs on the go-canary / experimental image, but that seems outside the scope of this PR.
> 
> @justaugustus, is there an issue tracking the fallout of the last CI go bump to 1.18 and how we'll change the rollout for go 1.19?

@BenTheElder @liggitt -- I've filed https://github.com/kubernetes/release/issues/2487 for this.

The tl;dr is:
- too many cooks
- we got caught off-guard by the go1.18 official release
- canary presubmits were not run on the PR updating to go1.18 official: https://github.com/kubernetes/kubernetes/pull/108870
- we bumped kubekins/krte variants (ref: https://github.com/kubernetes/test-infra/pull/25723) with looking at the canary presubmits

The timing/order of operations for much of this is documented in our [golang update issue template](https://github.com/kubernetes/release/blob/master/.github/ISSUE_TEMPLATE/dep-golang.md), so I think what it came down to here was a few too many folks working on this without following up on the previous steps.

/hold cancel

**Pull request merged**
